re.c

```diff
@@ -2283,8 +2283,14 @@ match_named_captures_iter(const OnigUChar *name, const OnigUChar *name_end,
     struct MEMO *memo = MEMO_CAST(arg);
     VALUE hash = memo->v1;
     VALUE match = memo->v2;
+    long symbolize = memo->u3.cnt;
 
     VALUE key = rb_enc_str_new((const char *)name, name_end-name, regex->enc);
+
+    if (symbolize > 0) {
+        key = rb_str_intern(key);
+    }
```

Comment:
@k-tsj I decide to re-use the existing iterator function (from `#named_captures`), since we need almost the same functionality for `#deconstruct_keys`. WDYT?

(And we can probably go further and `add `symbolize_keys: true` to `#named_captures`, which could be useful on its own)

Reply:
@palkan It looks good. (IMHO, `u3->state` is preferable to `u3->cnt`)

Reply:
Agree. Updated